src/Workspaces/Core/Portable/Shared/Extensions/ProjectExtensions.cs

```diff
@@ -53,5 +54,27 @@ internal static Project WithSolutionOptions(this Project project, OptionSet opti
 
         private static DocumentId? GetDocumentIdWithFilePath(this Project project, string filePath)
             => project.Solution.GetDocumentIdsWithFilePath(filePath).FirstOrDefault(id => id.ProjectId == project.Id);
+
+        public static Document GetRequiredDocument(this Project project, DocumentId documentId)
+        {
+            var document = project.GetDocument(documentId);
+            if (document == null)
+            {
+                throw new InvalidOperationException(WorkspaceExtensionsResources.The_solution_does_not_contain_the_specified_document);
+            }
+
+            return document;
+        }
+
+        public static TextDocument GetRequiredTextDocument(this Project project, DocumentId documentId)
```

Comment:
if we can use the getrequireddocument from the link mentioned above, we should probably move this method to the same file.

Reply:
Another Project extension method though. @dibarbet Do you really want these in ISolutionExtensions?